homeassistant/components/samsungtv/config_flow.py

```diff
@@ -78,15 +77,77 @@ def _get_entry(self):
             data=data,
         )
 
+    async def _async_set_device_unique_id(self, raise_on_progress=True):
+        """Set device unique_id."""
+
+        if self._id:
+            await self.async_set_unique_id(
+                self._id, raise_on_progress=raise_on_progress
+            )
+            self._abort_if_unique_id_configured()
+
+        if not self._device_info:
+            await self._async_get_and_check_device_info()
+
+        if uuid := self._device_info.get("device", {}).get(ATTR_UPNP_UDN.lower()):
+            self._id = uuid
+
+        if not self._id:
+            LOGGER.debug(
+                "Property " "%s" " is missing for host %s",
+                ATTR_UPNP_UDN.lower(),
+                self._host,
+            )
+            return self.async_abort(reason=RESULT_ID_MISSING)
+
+        if self._id.startswith("uuid:"):
+            self._id = self._id[5:]
+
+        await self.async_set_unique_id(self._id, raise_on_progress=raise_on_progress)
+        self._abort_if_unique_id_configured()
+
     def _try_connect(self):
```

Comment:
I think you could simplify this whole block to something like

```python

    def _try_connect(self):
        """Try to connect and check auth."""
        if self._bridge:
            return

        for method in SUPPORTED_METHODS:
            bridge = SamsungTVBridge.get_bridge(method, self._host)
            if self._bridge.try_connect() == RESULT_SUCCESS:
                self._bridge = bridge
                return
        LOGGER.debug("No working config found")
        raise data_entry_flow.AbortFlow(RESULT_CANNOT_CONNECT)

    def _connect_get_device_info(self):
        """Try connect and fetch the device info."""
        self._try_connect()
        return self._bridge.device_info

    async def _async_get_and_check_device_info(self):
        """Try to get the device info."""
        self._device_info = await self.hass.async_add_executor_job(self._connect_get_device_info)
```

Reply:
But I guess you want to avoid the `open`.  I'm not sure if you can get the unique id without it though